resources/models/availability.py

```diff
@@ -142,14 +142,17 @@ def _check_closed(self):
     def clean(self, ignore_overlap=False):
         super(Period, self).clean()
 
+        if self.start is None or self.end is None:
+            raise ValidationError(_("You must set 'start' and 'end' fields."), code="empty_start_end")
+
         if self.start > self.end:
             raise ValidationError("Period must start before its end", code="invalid_date_range")
 
-        self._validate_belonging()
         self._check_closed()
 
     def save(self, *args, **kwargs):
         ignore_overlap = kwargs.pop('ignore_overlap', False)
+        self._validate_belonging()
```

Comment:
Commit message just said there was problems with this, nothing about which kind of problems. Just wondering what effects this change might have since the validation should be done during the clean..

Could we have a look of those problems you had?